classes/app.rb

```diff
@@ -0,0 +1,70 @@
+require_relative 'music_album'
+require_relative 'genre'
+require './classes/preserve_data'
+
+class App
+  include ProcessData
+
+  def add_a_music_album(music_albums, genres)
+    puts 'Add a music album'
+    print 'Enter music albums name: '
+    name = gets.chomp
+    print 'Enter albums published date: '
+    publish_date = gets.chomp
+    print 'Is the album on spotify? (y/n)'
+    on_spotify = gets.chomp.downcase == 'y' || false
+    print 'Add genre: '
+    genre = genre_input(genres)
+    archived = false
+    album = MusicAlbum.new(publish_date, archived, name, on_spotify, genre)
+    music_albums.push(album)
+    album_data = { id: album.id, publish_date: publish_date, archived: false,
+                   name: name, on_spotify: on_spotify, genre: genre }
+    update_data('music', album_data)
+    puts "#{name} Album added successfully"
+    add_genre(genre, genres)
+  end
+
+  def genre_input(genres)
+    if genres.empty?
+      puts 'No Genres have been added yet'
+      print 'enter a Genre name:'
+    else
+      puts 'Select genre for the list.'
+      genres.each_with_index { |genre, index| puts "#{index} Name: #{genre.name}" }
+      print 'Enter name of genre or Enter a new Genre: '
+    end
+    gets.chomp.downcase
+  end

```

Comment:
you should create a different method for getting inputs because this method add_a_music_album is only responsible for adding the music album. Then, It would be according to the SOLID rules.